### PR TITLE
[DEV-1207] beta feedback

### PR DESCRIPTION
**Pull request by Krysh90**

### CODE REVIEW

**Comment by Krysh90** [comment]
removed portfolio balance, due to only showing main token and not all possible tokens

File: .env.dev [hunk 1]
```diff
@@ -1,4 +1,5 @@
 REACT_APP_API_URL=https://dev.api.dfx.swiss/v1
+REACT_APP_DFX_URL=https://dfx.swiss/
```
Comment (davidleomay) [inline .env.dev:2]: You could add a `.env` file, which will act as a base file. So `REACT_APP_DFX_URL` could be moved there.

File: src/components/tabs/buy-tab-content/buy.process.tsx [hunk 2]
```diff
@@ -68,6 +69,13 @@ export function BuyTabContentProcess({ asset, onBack }: BuyTabContentProcessProp
     ['GBP']: 'British Pound',
   };
 
+  const currencySymbol: Record<string, string> = {
+    ['EUR']: '€',
+    ['USD']: '$',
+    ['CHF']: '₣',
+    ['GBP']: '£',
+  };
```
Comment (davidleomay) [inline src/components/tabs/buy-tab-content/buy.process.tsx:77]: Maybe move to fiat.hook?

File: src/api/hooks/api.hook.ts [hunk 3]
```diff
@@ -10,6 +10,12 @@ export interface CallConfig {
   method: 'GET' | 'PUT' | 'POST' | 'DELETE';
   data?: any;
   noJson?: boolean;
+  specialHandling?: SpecialHandling;
+}
+
+interface SpecialHandling {
+  action?: () => void;
```
Comment (davidleomay) [inline src/api/hooks/api.hook.ts:17]: `action` should not be nullable 🤔

File: src/api/hooks/user.hook.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { useApi } from './api.hook';
 
 export interface UserInterface {
   getUser: () => Promise<User | undefined>;
-  changeUser: (user?: User) => Promise<User | undefined>;
+  changeUser: (user?: User, specialHandling?: () => void) => Promise<User | undefined>;
```
Comment (davidleomay) [inline src/api/hooks/user.hook.ts:6]: Rename `specialHandling` to something like `userLinkAction`?

File: src/api/contexts/user.context.tsx [hunk 5]
```diff
@@ -49,13 +51,17 @@ export function UserContextProvider(props: PropsWithChildren): JSX.Element {
   async function changeMail(mail: string): Promise<void> {
     if (!user) return; // TODO (Krysh) add real error handling
     setIsUserUpdating(true);
-    return changeUser({ ...user, mail })
+    return changeUser({ ...user, mail }, userLinkAction)
```
Comment (davidleomay) [inline src/api/contexts/user.context.tsx:54]: Why not add the callback as parameter to changeMail?
```ts
async function changeMail(mail: string, onUserLink: () => void): Promise<void> {
```

Comment (davidleomay) [reply src/api/contexts/user.context.tsx:54]: Ah because it is called from different locations?

Comment (davidleomay) [reply src/api/contexts/user.context.tsx:54]: Maybe rename the `register` to something more meaningful then?

Comment (Krysh90) [reply src/api/contexts/user.context.tsx:54]: we need the functionality just once from the main screen. Then all other occurrences of email edit leading to 202 would trigger the popup, which is very neat. As I am not forced to add the same functionality to every change email functionality.

Regarding the name it is registering it once and I wanted to keep it plain as we are using a same naming for metamask hook. What if a second hook is added, would we add a new function or add a new parameter to the `register` function. I would keep it like that

File: .env.dev (hunk 1 above)
Comment (Krysh90) [reply .env.dev:2]: is it really needed to move it? I don't mind code duplication in an `.env` file

File: src/api/contexts/user.context.tsx (hunk 5 above)
Comment (davidleomay) [reply src/api/contexts/user.context.tsx:54]: Ok agreed 👍

File: .env.dev (hunk 1 above)
Comment (davidleomay) [reply .env.dev:2]: No, just a proposition in case you didn't know.

**Review by davidleomay — APPROVED** [review]

File: .env.dev (hunk 1 above)
Comment (Krysh90) [reply .env.dev:2]: didn't know, indeed :D

**Review by davidleomay — APPROVED** [review]

**Pull request merged**